docs/source/changelog.md

```diff
@@ -12,6 +12,326 @@ and as we merge [breaking changes in pull
+of `singleuser.networkPolicy.egress` to be less permissive, you should consider
+disabling all `singleuser.networkPolicy.egressAllowRules` like this
+to not risk escalating the permissions.
+
+```yaml
+singleuser:
+  networkPolicy:
+    egressAllowRules:
+      cloudMetadataServer: false
+      dnsPortsPrivateIPs: false
+      nonPrivateIPs: false
+      privateIPs: false
+```
+
+For more details, see the documentation on [Kubernetes Network Policies](netpol)
+and the configuration reference entries under
+[`...networkPolicy.egress`](schema_hub.networkPolicy.egress) and
+[`...networkPolicy.egressAllowRules`](schema_hub.networkPolicy.egressAllowRules).
+
+#### Upgrade instructions
+
+1. Adjust your configuration based on the security notice above if it applies to
+   your deployment.
+2. Prepare a user image with a modern version of JupyterHub installed. Install
+   `jupyterhub>=2.3.1` via pip or `jupyterhub-base>=2.3.1` via conda/mamba in
+   the image.
+   Update the configuration to use this new image, which is typically done via
+   `singleuser.image` or as part of `singleuser.profileList`.
+3. Read and adjust your configuration to the breaking changes below.
+
+#### Breaking changes
+
+These breaking changes have been made relative to the 1.\* series of Z2JH releases.
+
+##### JupyterHub 2 and related hub components
+
+Z2JH 2.0.0 upgrades JupyterHub to the 2.\* series, and also upgrades all hub components.
+If you are using any custom JupyterHub services, addons, API integrations, or extra configuration, you should review the breaking changes in the
+[JupyterHub 2.0.0 changelog](https://github.com/jupyterhub/jupyterhub/blob/2.3.1/docs/source/changelog.md#200).
+
+JupyterHub 2 uses an updated database schema.
+Z2JH 2.0.0 automatically handles the upgrade, but it will not be possible to downgrade to older releases after this.
+
+JupyterHub 2 adds RBAC for managing permissions in JupyterHub.
+The old permissions model of admin/non-admin still works but you should use [RBAC to assign the required privileges to users or services in future](https://jupyterhub.readthedocs.io/en/stable/rbac/index.html)
+
+See
+`TODO: link to Notable dependencies updated`
+for more information on other upgraded hub components.
+
+##### JupyterLab and Jupyter Server
+
+The default singleuser server is [JupyterLab](https://jupyterlab.readthedocs.io/), running on [Jupyter server](https://jupyter-server.readthedocs.io/en/latest/).
+To switch back to Jupyter Notebook either configure/rebuild your singleuser image to default to notebook, or see [the documentation on user interfaces](user-interfaces)
+
+##### Default to using the container image's command instead of `jupyterhub-singleuser` [#2449](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/2449)
+
+Z2JH now launches the container's default command (equivalent to setting `CMD` in a `Dockerfile`) instead of overriding it. To restore the old bvehaviour set
```

Comment:
```suggestion
Z2JH now launches the container's default command (as set e.g. by `CMD` in a `Dockerfile`) instead of overriding it. To restore the old behaviour, set
```